torchdynamo/eval_frame.py

```diff
@@ -322,12 +322,66 @@ def toy_example(a, b):
     backend_ctx_ctor = getattr(backend, "backend_ctx_ctor", null_context)
 
     if nopython:
-        return optimize_assert(backend, guard_export_fn=None)
+        return optimize_assert(backend, guard_export_fn=guard_export_fn)
     return _optimize_catch_errors(
-        convert_frame.convert_frame(backend, guard_export_fn=None), backend_ctx_ctor
+        convert_frame.convert_frame(backend, guard_export_fn=guard_export_fn),
+        backend_ctx_ctor,
     )
 
 
+def explain(f, *args, **kwargs):
+    import torchdynamo.symbolic_convert as symbolic_convert
+
+    symbolic_convert.explain = True
```

Comment:
Use patch so if we throw an exception you don't leak the old value

Reply:
sg